### PR TITLE
TiDB supports `MID(str, pos[, len])`

### PR DESCRIPTION
**Pull request by dveeden**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

https://github.com/pingcap/tidb/pull/55541

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Defined2014 — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@dveeden https://github.com/dveeden/docs/pull/8 PTAL

**Comment by dveeden** [comment]
> @dveeden [dveeden#8](https://github.com/dveeden/docs/pull/8) PTAL

Merged

**Review by Oreoxmt — APPROVED** [review]

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -1445,11 +1445,11 @@ SELECT MAKE_SET(b'111','foo','bar','baz');
 
 ### [`MID()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_mid)
 
-The `MID(str, pos, len)` function returns a substring starting from the specified `pos` position with the `len` length.
+The `MID(str, pos[, len])` function returns a substring starting from the specified `pos` position with the `len` length.
 
-If any of the arguments are `NULL`, the function returns `NULL`.
+Starting from v8.4.0, TiDB supports the two-argument variant `MID(str, pos)`. If `len` is omitted, this function returns all remaining characters from the specified `pos` position to the end of the string.
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1450]:
```suggestion
Starting from v8.4.0, TiDB supports the two-argument variant `MID(str, pos)`. If `len` is not specified, this function returns all remaining characters from the specified `pos` position to the end of the string.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**